operator/runner.go

```diff
@@ -0,0 +1,361 @@
+package operator
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+
+	"github.com/grafana/grafana-app-sdk/app"
+	"github.com/grafana/grafana-app-sdk/k8s"
+	"github.com/grafana/grafana-app-sdk/metrics"
+	"github.com/grafana/grafana-app-sdk/resource"
+)
+
+// Runner runs an app.App as a standalone operator, capable of exposing admission (validation, mutation)
+// and conversion as webhooks, and running a main control loop with reconcilers and watchers.
+// It relies on the Kinds managed by the app.App already existing in the API server it talks to, either as CRD's
+// or another type. It does not support certain advanced app.App functionality which is not natively supported by
+// CRDs, such as arbitrary subresources (app.App.CallSubresource). It should be instantiated with NewRunner.
+type Runner struct {
+	config        RunnerConfig
+	webhookServer *webhookServerRunner
+	metricsServer *metricsServerRunner
+	startMux      sync.Mutex
+	running       bool
+	runningWG     sync.WaitGroup
+}
+
+// NewRunner creates a new, properly-initialized instance of a Runner
+func NewRunner(cfg RunnerConfig) (*Runner, error) {
+	// Validate the KubeConfig by constructing a rest.RESTClient with it
+	// TODO: this requires a GroupVersion, which gets set up based on the kind
+	// _, err := rest.RESTClientFor(&cfg.KubeConfig)
+	// if err != nil {
+	// 	return nil, fmt.Errorf("invalid KubeConfig: %w", err)
+	// }
```

Comment:
I think just validating that it has things like user / token / api address / certs / etc. is enough.

I was under the impression that there was a validation function like that somewhere in `client-go` but that doesn't seem to be the case.

I suggest we leave this as is for now and write a function like that later.